src/Chullo.php

```diff
@@ -58,56 +62,49 @@ public function getBaseUri() {
      * @param array     $headers        HTTP Headers
      * @param string    $transaction    Transaction id
      *
-     * @return mixed    String or binary content if 200. Null if 304.
+     * @return mixed    Full response if found.  Null otherwise.
      */
-    public function getResource($uri = "",
-                                $headers = [],
-                                $transaction = "") {
-        // Set headers
-        $options = ['headers' => $headers];
-
-        // Ensure uri takes transaction into account.
-        $uri = $this->prepareUri($uri, $transaction);
-
-        // Send the request.
-        $response = $this->client->request(
-            'GET',
+    public function getResource(
+        $uri = "",
+        $headers = [],
+        $transaction = ""
+    ) {
+        $response = $this->api->getResource(
             $uri,
-            $options
+            $headers,
+            $transaction
         );
-
-        $code = $response->getStatusCode();
-
-        if ($code == 304) {
+        if ($response->getStatusCode() != 200) {
             return null;
         }
 
-        return $response->getBody();
+        return $response;
```

Comment:
Ha, yeah.  So what you're referring to there is my horrible abuse of PHP interfaces.  They don't actually take return type into consideration.  So I just extended IFedoraClient from IFedoraApi and just noted the different return types in the comments.  I'm totally aware of how that's probably misleading.  I guess just make a completely seperate interface for IFedoraClient, then?  I did purposefully like keeping the names the same since we're really just wrapping the Api with the Client.

As for needing mimetype (or other things), I figured you'd most likely know before hand when making the GET request.  If you didn't know what the resource was, you could either use a HEAD request to find out, or hit the fcr:metadata resource in order to get the technical details on the file.  Perhaps having a function that would return a simple struct, with both the body content and the fcr:metadata as an Easy_RdfGraph would be useful?